app/src/org/odk/collect/android/views/media/AudioButton.java

```diff
@@ -362,24 +278,83 @@ private String formatTime(Integer milliseconds) {
         return returnValue;
     }
 
-    public void startPlaying() {
-        logAction("start");
-        controller.playCurrentMediaEntity();
-        setStateToPlaying();
-    }
+    public AudioController buildAudioControllerInstance() {
+        return new AudioController() {
+            private MediaPlayer mp;
+            boolean alive = false;
 
-    public void endPlaying() {
-        logAction("stop");
-        if (!currentState.equals(MediaState.Playing)) {
-            controller.releaseCurrentMediaEntity();
-            setStateToReady();
-        }
-    }
+            @Override
+            public MediaEntity getCurrMedia() {
+                return null;
+            }
 
-    public void pausePlaying() {
-        logAction("pause");
-        controller.pauseCurrentMediaEntity();
-        setStateToPaused();
-    }
+            @Override
+            public void setCurrent(MediaEntity newEntity) {
+                this.mp = newEntity.getPlayer();
+            }
+
+            @Override
+            public void setCurrent(MediaEntity newEntity, AudioButton newButton) {
+                setCurrent(newEntity);
+            }
+
+            @Override
+            public void releaseCurrentMediaEntity() {
+                if (mp != null) {
+                    mp.reset();
+                    mp.release();
+                    mp = null;
```

Comment:
adding "mp = null" here fixes the problem where audio doesn't stop when you advance the form past the audio playing view.